spec/models/profile_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+describe Profile do
+  let(:user) { create(:user, :signed_up) }
+  let(:profile) do
+    Profile.create(
+      user_id: user.id
+    )
+  end
+
+  subject { profile }
+
+  it { is_expected.to belong_to(:user) }
+
+  describe 'allows only one active Profile per user' do
+    it 'prevents create! via ActiveRecord uniqueness validation' do
+      profile.active = true
+      profile.save!
+      expect do
+        Profile.create!(user_id: user.id, active: true)
+      end.to raise_error(ActiveRecord::RecordInvalid)
+    end
+
+    it 'prevents save! via psql unique partial index' do
+      profile.active = true
```

Comment:
I am looking for a specific exception class, which requires `save!` (not `save`) to trigger.